### PR TITLE
Added PSD plotting function and example notebook

### PR DESCRIPTION
**Pull request by MerlinDumeur**

### CODE REVIEW

File: mfanalysis/psd.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+                    nperseg=window_size,
+                    noverlap=window_size/2,
+                    nfft=4096,
+                    detrend=False,
+                    return_onesided=True,
+                    scaling='density',
+                    average='mean',
+                    fs = 2*np.pi)
+
+    f_Fourier = fs * np.linspace(0,0.5, window_size / 2 + 1)
```
Comment (philouc) [inline mfanalysis/psd.py]: PEP8 0, 0.5

File: mfanalysis/psd.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    f_Fourier, pds_Fourier = np.log2(f_Fourier[1:]), np.log2(pds_Fourier[1:]) + 2
+
+    mfa = MFA(wt_name=f'db2',
+                 formalism='wcmf',
+                 j1=1,j2=13,
```
Comment (philouc) [inline mfanalysis/psd.py]: Maybe one line per parameter: split j1= and j2=

File: mfanalysis/psd.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
+from scipy.signal import welch
+import numpy as np
+import matplotlib.pyplot as plt
+
+from .mfa import MFA
+
+def plot_psd(signal, fs):
+
+    overlapingratio = 4
```
Comment (philouc) [inline mfanalysis/psd.py]: better var name: overlap_ratio

File: mfanalysis/psd.py (hunk 1 above)
Comment (philouc) [inline mfanalysis/psd.py]: freq_Fourier instead of f_Fourier (more explicit name for ease of legibility)

File: mfanalysis/psd.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+                    noverlap=window_size/2,
+                    nfft=4096,
+                    detrend=False,
+                    return_onesided=True,
+                    scaling='density',
+                    average='mean',
+                    fs = 2*np.pi)
+
+    f_Fourier = fs * np.linspace(0,0.5, window_size / 2 + 1)
+    f_Fourier, pds_Fourier = np.log2(f_Fourier[1:]), np.log2(pds_Fourier[1:]) + 2
```
Comment (philouc) [inline mfanalysis/psd.py]: propagate changes

File: mfanalysis/psd.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    mfa.wavelet_coeffs = None
+    mfa.wavelet_leaders = None
+    mfa.structure = None
+    mfa.cumulants = None
+
+    mfa._wavelet_analysis(signal)
+
+    pds_wavelet = [np.square(arr).mean() for arr in mfa.wavelet_coeffs.values.values()]
+    
+    j_wavelet = np.arange(len(pds_wavelet))+1
```
Comment (philouc) [inline mfanalysis/psd.py]: PEP8 white space before and after + 1

File: mfanalysis/psd.py [hunk 6]
```diff
@@ -4,26 +4,60 @@
 
 from .mfa import MFA
 
-def plot_psd(signal, fs):
+def plot_psd(signal, fs, n_fft=4096, segment_size=None, n_moments=2):
+    """
+    Plot the superposition of Fourier-based Welch estimation and Wavelet-based estimation of PSD on a log-log graphic.
+    Based on the `wavelet_fourier_spectrum` function of the MATLAB mf_bs_toolbox-v0.2
```
Comment (philouc) [inline mfanalysis/psd.py]: Provide web reference to the MATLAB functions, name explicitly Herwig's toolbox location

File: mfanalysis/psd.py [hunk 7]
```diff
@@ -4,26 +4,60 @@
...
+    
+    segment_size: int | None
+        Length of Welch segments.
+        Defaults to None, which sets it equal to `n_fft`
+
+    overlapingratio: int
+        Defined as ceil(log2(len(signal))) / log2(segment_size)
+
+    n_moments: int
+        Number of vanishing moments of the Daubechies wavelet used in theWavelet decomposition.
```
Comment (philouc) [inline mfanalysis/psd.py]: typo ie white space missing theWavelet -> the Wavelet

File: mfanalysis/psd.py [hunk 8]
```diff
@@ -79,20 +79,18 @@ def plot_psd(signal, fs, n_fft=4096, segment_size=None, n_moments=2):
 
     mfa._wavelet_analysis(signal)
 
-    pds_wavelet = [np.square(arr).mean() for arr in mfa.wavelet_coeffs.values.values()]
+    psd_wavelet = [np.square(arr).mean() for arr in mfa.wavelet_coeffs.values.values()]
 
-    j_wavelet = np.arange(len(pds_wavelet)) + 1
+    j_wavelet = np.arange(len(psd_wavelet)) + 1
     freq_wavelet = (3/4 * fs) / (np.power(2, j_wavelet))
```
Comment (philouc) [inline mfanalysis/psd.py]: Perhaps it's safer to write 3. /4 to be sure the result is real valued, isn't it?

Comment (MerlinDumeur) [reply mfanalysis/psd.py]: In python 3, the / operator is float division so it should not be a problem

Comment (MerlinDumeur) [reply mfanalysis/psd.py]: I wrote the code for python 3, and I didn't plan to support python 2.
There are good reasons for this, see https://python3statement.org/ (most datascience libraries including scikit-learn and MNE are participating)

Comment (philouc) [reply mfanalysis/psd.py]: Sorry about that, You're right. I missed that point

**Pull request merged**